### ISSUE TITLE
acs-engine upgrade removed masters and returned varible limit exceeded error.

### ISSUE THREAD

**Issue by BIGPHATTOBY** [Azure/acs-engine#3528]
**Is this a request for help?**:
Yes.
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
Issue
---

**What version of acs-engine?**:
Version: v0.20.0
---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes 

**What happened**:
Running out of IPs in my vnet returned a nil panic, for better troubleshooting information I initiated an upgrade from 1.10 to 1.11, which resulted in:
```
Error upgrading cluster: resources.DeploymentsClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="InvalidTemplate" Message="Deployment template validation failed: 'The number of template variables limit exceeded. Limit: '256' and actual: '261'. Please see https://aka.ms/arm-template/#variables for usage details.'."
```
The error occurred after acs-engine deleted my masters, effectively rendering the cluster unmanageable.

**What you expected to happen**:
For the cluster to a) get upgraded or b) fail without losing my masters.

**How to reproduce it** (as minimally and precisely as possible):
The clusters is deployed with following config:
3 masters
66 windows nodes in 1. nodepool
18 windows nodes in 2. nodepool
4 linux nodes in a 3. nodepool
version 1.10

Deploy with acs-engine v0.18.0 and try to upgrade with acs-engine v0.20.0 to 1.11.1

Please ask for any logs you could need to debug this issue, i cannot fetch any master logs, since well, there are no masters anymore.

rerunning the acs-engine upgrade command at looking at the output, it seems that acs-engine is aware that there is no masters available and is trying to re-create a master, then is when the variable limit exceeded error is returned.

**Comment by jackfrancis** [comment]
@BIGPHATTOBY This is a known limitation of using many nodes in > 1 agentpool (i.e., the number of template variables scales linearly with nodes + pools)

You could try manually removing a nodepool after a cordon/drain, if that is possible given your current workloads, or reducing node count.

Again, this is a undesirable situation, sorry about that. Getting yourself out of it without building a new cluster will be manual. We'll try to help.

**Comment by jackfrancis** [comment]
Fixed with #3649
